contracts/RiteOfMoloch.sol

```diff
@@ -343,27 +334,23 @@ contract RiteOfMoloch is InitializationData, ERC721Upgradeable, AccessControlUpg
 
         // the total amount of blood debt
         uint256 total;
-
-        for (uint256 i = 0; i < _failedInitiates.length; ++i) {
+        uint i;
+        
+        while (i < _failedInitiates.length) {
             // store each initiate's address
             address initiate = _failedInitiates[i];
-
-            // access each initiate's starting time
-            uint256 deadline = deadlines[initiate];
-
-            if (block.timestamp > deadline && _staked[initiate] > 0) {
-
-                // access each initiate's balance
-                uint256 balance = _staked[initiate];
-
+            // increments i
+            unchecked { ++i; }
+            
+            if (block.timestamp > deadlines[initiate] && ! isMember(initiate)) {
                 // calculate the total blood debt
-                total += balance;
+                total += staked[initiate];
```

Comment:
Also, shouldn't we add a `require(member.exists)` (for `msg.sender`) in the `sacrifice` function to prevent the critical vulnerability of slashing an address that became a member but hasn't claimed its stake?

Reply:
Overall gas change: 870 (0.070%)(+)
Can dig deeper if you think it's worth it.